src/classes/Math/Quaternion.js

```diff
@@ -100,6 +104,95 @@ Goblin.Quaternion.prototype = {
 		dest.z = iz * qw + iw * -qz + ix * -qy - iy * -qx;
 	},
 
+	setFromMat4: function (m) {
+        var m00, m01, m02, m10, m11, m12, m20, m21, m22,
+            tr, s, rs, lx, ly, lz;
+
+        // Cache matrix values for super-speed
+        m00 = m.e00;
+        m01 = m.e10;
+        m02 = m.e20;
+        m10 = m.e01;
+        m11 = m.e11;
+        m12 = m.e21;
+        m20 = m.e02;
+        m21 = m.e12;
+        m22 = m.e22;
+
+        // Remove the scale from the matrix
+        lx = 1 / Math.sqrt(m00 * m00 + m01 * m01 + m02 * m02);
+        ly = 1 / Math.sqrt(m10 * m10 + m11 * m11 + m12 * m12);
+        lz = 1 / Math.sqrt(m20 * m20 + m21 * m21 + m22 * m22);
+
+        m00 *= lx;
+        m01 *= lx;
+        m02 *= lx;
+        m10 *= ly;
+        m11 *= ly;
+        m12 *= ly;
+        m20 *= lz;
+        m21 *= lz;
+        m22 *= lz;
+
+        // http://www.cs.ucr.edu/~vbz/resources/quatut.pdf
+
+        tr = m00 + m11 + m22;
+        if (tr >= 0) {
```

Comment:
Ni idea what happens below, just drawing your attention, there is no zero division checks as well.

Reply:
Please refer to my previous comment.